app/models/internal/scheduling/Job.java

```diff
@@ -61,5 +69,6 @@
      * @return A {@link CompletionStage} that completes with the job's result, or with the exception the job encounters (if any).
      */
     CompletionStage<? extends T> execute(Injector injector, Instant scheduled);
+    // ^ TODO(spencerpearson): does this want to be a CompletableFuture now that we allow timeouts?
```

Comment:
Please address.

Reply:
I would like to hold off on the signature changes / implementation of timeouts for a later diff: it's nontrivial (as you'll hear at standup today :P).